scipy/optimize/_numdiff.py

```diff
@@ -285,18 +303,19 @@ def approx_derivative(fun, x0, method='3-point', rel_step=None, abs_step=None,
                      analytically continued to the complex plane. Otherwise,
                      produces bogus results.
     rel_step : None or array_like, optional
-        Relative step size to use. The absolute step size is computed as
-        ``h = rel_step * sign(x0) * max(1, abs(x0))``, possibly adjusted to
-        fit into the bounds. For ``method='3-point'`` the sign of `h` is
-        ignored. If None (default) then step is selected automatically,
-        see Notes.
+        Relative step size to use. If None (default) the absolute step size is
+        computed as ``h = rel_step * sign(x0) * max(1, abs(x0))``, with
+        `rel_step` being selected automatically, see Notes. Otherwise
+        ``h = rel_step * sign(x0) * abs(x0)``. For ``method='3-point'`` the
+        sign of `h` is ignored. The calculated step size is possibly adjusted
+        to fit into the bounds.
```

Comment:
```suggestion
        to fit into the bounds, and the automatic step size is always used
        when ``abs(x0) == 0``.
```
Or some other adjustment that makes it clear that it doesn't always use the supplied `rel_step`

Reply:
I've improved the notes to make this clearer. `approx_derivative` is still a private function, but it's good to have good docstrings.